tests/test_browser.py

```diff
@@ -4716,3 +4716,45 @@ def test_minimal_runtime_loader_shell(self):
       for modularize in [[], ['-s', 'MODULARIZE=1']]:
         print(str(args + wasm + modularize))
         self.btest('minimal_hello.c', '0', args=args + wasm + modularize)
+
+  def _test_es6(self, args):
+    args = args + ['-s', 'EXPORT_ES6', '-s', 'MODULARIZE', '-s', 'MODULARIZE_INSTANCE', '-s', 'ASSERTIONS', '-o', 'test.mjs']
+    print(args)
+    create_test_file('src.c', self.with_report_result(open(path_from_root('tests', 'browser_test_hello_world.c')).read()))
+    self.compile_btest(['src.c'] + args)
+    create_test_file('test.html', '''
+      <script>
+        var logs = 0;
+        function log(e) {
+          var xhr = new XMLHttpRequest();
+          xhr.open('GET', encodeURI('http://localhost:8888?stdout=' + e + ' : ' + new Error().stack + ' : ' + logs++));
+          xhr.send();
+        }
+        log('startup');
+        window.onerror = function(e) {
+          log(e);
+        };
+      </script>
+      <script type="module" src="test.mjs"></script>
+      <script>
+        log('after');
+      </script>
+    ''')
+    self.run_browser('test.html', None, '/report_result?0')
+
+  def test_es6_O0(self):
+    self._test_es6([])
+
+  def test_es6_O3(self):
+    self._test_es6(['-O3'])
+
+  def test_es6_async(self):
+    self._test_es6(self.get_async_args())
+
+  @requires_threads
+  def test_es6_threads(self):
+    # TODO: use PROXY_TO_PTHREAD and/or '-s', 'PTHREAD_POOL_SIZE=1' once https://bugs.chromium.org/p/chromium/issues/detail?id=680046 is fixed
```

Comment:
This should work on Chrome if `--enable-experimental-web-platform-features` is set (sorry, forgot to mention that).
The Firefox bug is also: https://bugzilla.mozilla.org/show_bug.cgi?id=1540913